### DISCUSSION TITLE
Gothic engine documentation

### DISCUSSION THREAD

**Discussion by CrayolaEater** [Try/OpenGothic#257] [Q&A]
Hello! So I'm doing some reverse engineering for the gothic 2 game and I was wondering if you can provide some documentation or info to help me speed things up. Any help would be appreciated ! ;)

**Comment by PolyMeilex** [comment]
This should be enough to get started: https://ataulien.github.io/Inside-Gothic/
It has tones of interesting info.
It used to be a part of ReGoth project once that was a thing.

**Reply by CrayolaEater** [reply]
Yes, this is useful as an arhitectural point of view to see how the engine works, but I need specific info for example: where the X function is stored in memory (exact address) what does Y instruction does, that sort of thing. Im doing a lot of investigation and this could save me a lot of time.

**Reply by ErrorFlexXx** [reply]
Like this one here, [https://raw.githubusercontent.com/Siemekk/AST-World-Editor/master/G2_names.txt](https://raw.githubusercontent.com/Siemekk/AST-World-Editor/master/G2_names.txt) I guess.
Don't know, if it's from G2 addon or the first G2 version.
Sadly, there is no documentation regarding the function descriptions, but projects that detour functions to do something can be found.
Some info can be found with googling the specific functions.

**Comment by Try** [comment]
Here we don't do actual source-code reverse-engineering, since there is little value in it. We do only file-format reverse-engineering and implementation of script-api entry-points.

For "here the X function is stored in memory (exact address) what does Y instruction does", sort of things you can take a look at Ikarus:
https://github.com/Lehona/Ikarus/tree/master/EngineClasses_G2

**Reply by CrayolaEater** [reply]
what do you mean file-format RE and script-api entry-points? there is an alternative rather than digging through the disassembled code?

**Reply by CrayolaEater** [reply]
Also what is that link about? I don't exactly understand

**Reply by PolyMeilex** [reply]
> what do you mean file-format RE and script-api entry-points? there is an alternative rather than digging through the disassembled code?

It basically means that if you are writing your own engine, you don't have to dig through assembly that much, you only need reimplementation of file formats, and script API endpoints. So the alternative here is writing an engine from scratch.

  

> Also what is that link about? I don't exactly understand

Ikarus uses memory hacking, so you can just take addresses to functions from its source code. You could also look at Union source code, as it does this as well.
